### PR TITLE
rpc: Add AccountMixerService

### PR DESCRIPTION
**Pull request by alexlyp**
This adds a new rpc Service for account mixing which uses the newly implemented CoinShuffle++ mixer.  

The request RunAccountMixerRequest contains the following fields:

MixedAccount: The account in which mixed funds will end up.
MixedAccountBranch: The account branch in which the mixed funds will end up.
ChangeAccount:  Any unmixed change will be put into this account.
CsppServer: The URL:port for your desired cspp mixing server.

### CODE REVIEW

File: internal/rpc/rpcserver/server.go [hunk 1]
```diff
@@ -2152,6 +2162,60 @@ func (s *loaderServer) checkReady() bool {
 	return atomic.LoadUint32(&s.ready) != 0
 }
 
+// StartAccountMixerService starts the AccountMixerService.
+func StartAccountMixerService(server *grpc.Server, loader *loader.Loader) {
```
Comment (matheusd) [inline internal/rpc/rpcserver/server.go]: You need to add a call for this somewhere, otherwise we can't run mixes.

**Comment by jrick** [comment]
> Also I tried using `https://cspp.decred.org:15760` as mixer address but didn't get any mixes done (nor any `Dialing cspp...` log messages. Are we running a testnet cspp server?

The mixing server does not serve HTTPS (except for the web frontend).  It uses a custom messaging protocol over TLS.  The server name should be just `cspp.decred.org:15760`.

File: internal/rpc/rpcserver/server.go [hunk 2]
```diff
@@ -2152,6 +2162,60 @@ func (s *loaderServer) checkReady() bool {
...
+
+	tb := ticketbuyer.New(wallet)
+
+	// Set ticketbuyerV2 config
+	tb.AccessConfig(func(c *ticketbuyer.Config) {
+		c.MixedAccountBranch = req.MixedAccountBranch
+		c.MixedAccount = req.MixedAccount
+		c.ChangeAccount = req.ChangeAccount
+		c.CSPPServer = req.CsppServer
+		c.BuyTickets = false
```
Comment (matheusd) [inline internal/rpc/rpcserver/server.go]: Seems you also need to specify `MixChange = true`.

After doing these changes and using the server jrick specified the mixes are being done.

**Comment by chappjc** [comment]
@matheusd Also note that the testnet server that jrick pointed requires no external x509 cert file as it's coded into dcrwallet.

**Comment by alexlyp** [comment]
> We don't currently enforce a single ticketbuyer instance in the server and though decrediton doesn't run multiple instances of it maybe we should limit (in the server) a single mixer + a single ticketbuyer instance?
> 
> I'm considering whether someone could start both at the same time and cause address reuse through that.
> 

We don't currently enforce a single ticketbuyer instance in the server and though decrediton doesn't run multiple instances of it maybe we should limit (in the server) a single mixer + a single ticketbuyer instance?

I'm considering whether someone could start both at the same time and cause address reuse through that.

Also I tried using https://cspp.decred.org:15760 as mixer address but didn't get any mixes done (nor any Dialing cspp... log messages. Are we running a testnet cspp server?

> Also I tried using `https://cspp.decred.org:15760` as mixer address but didn't get any mixes done (nor any `Dialing cspp...` log messages. Are we running a testnet cspp server?

Yah it's worth discussing using a single instance.   Reasonable to be concerned about address reuse, open to whatever you and @jrick think.

**Pull request merged**